### PR TITLE
Add support for secondary tab bar

### LINKED ISSUE #111962: Update `TabBar` to support Material 3
| M2 | M3 |
| --------------- | --------------- |
|  |  |

We should update the `TabBar` widget to support this Material 3 look and [tokens](https://github.com/flutter/flutter/blob/master/dev/tools/gen_defaults/data/navigation_tab_primary.json).

Specs: https://m3.material.io/components/tabs/specs

### PR DESCRIPTION
**Pull request by guidezpl**
## Description

* Introduce `secondary` constructor for `TabBar`
* Generate defaults from design tokens for secondary tab bar
* Uses correct defaults throughout
* Adds a M3 sample showcasing nested (primary + secondary) tabs
    image

## Related Issues
 - Fixes https://github.com/flutter/flutter/issues/111962

## Tests
 - TODO

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by TahaTesser** [comment]
Thanks for pushing this! I'll add the tests and get second review and land this soon.

File: packages/flutter/lib/src/material/tabs.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/tabs.dart:194]:
```suggestion
  TabBarTheme _defaults(BuildContext context) {
    if (Theme.of(context).useMaterial3) {
      return isPrimary
        ? _TabsPrimaryDefaultsM3(context)
        : _TabsSecondaryDefaultsM3(context);
    }
    return _TabsDefaultsM2(context);
  }
```

File: packages/flutter/lib/src/material/tabs.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/tabs.dart:712]:
```suggestion
  }) : _isPrimary = true,
       assert(indicator != null || (indicatorWeight > 0.0));
```

File: packages/flutter/lib/src/material/tabs.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/tabs.dart:761]:
```suggestion
  }) : _isPrimary = false,
       assert(indicator != null || (indicatorWeight > 0.0));
```

File: packages/flutter/test/material/tab_bar_theme_test.dart
Comment (Piinks) [inline packages/flutter/test/material/tab_bar_theme_test.dart:166]:
```suggestion
    // Test default labelPadding.
```

**Review by Piinks — COMMENTED** [review]
Mostly some style nits, but over all this is nice!

File: packages/flutter/lib/src/material/tabs.dart
Comment (Piinks) [reply packages/flutter/lib/src/material/tabs.dart:1102]: I was able to resolve the style bits in the github UI, so only this is remaining.
Also, are there golden file tests? Or are those in https://github.com/flutter/flutter/pull/123127?

Comment (TahaTesser) [reply packages/flutter/lib/src/material/tabs.dart:1102]: Yep, I will add golden tests  in https://github.com/flutter/flutter/pull/123127 as that PR fixes divider issue  and produces golden changes.  

My goal with this PR is to be as safe as possible so we can have seocndary tab regardless of the golden changes.

**Comment by TahaTesser** [comment]
> Mostly some style nits, but over all this is nice!

Thank you so much for adding fixing the style!

File: packages/flutter/lib/src/material/tabs.dart
Comment (TahaTesser) [reply packages/flutter/lib/src/material/tabs.dart:1102]: I did do a small cleanup in the test classes for this PR.

Comment (Piinks) [reply packages/flutter/lib/src/material/tabs.dart:1098]: Why not just pass the value of this getter to _TabStyle then as a constructor parameter? It is only created by the _TabBarState in build & _buildStyledTab. :)

**Review by Piinks — APPROVED** [review]
Nice! LGTM, thank you!

**Pull request merged**
